### PR TITLE
Fix waypoint orientation return order and change joint waypoint to eigen

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@mpowelson I made some changes to so the config only require base class and changed joint waypoint to use an eigen vector instead of std::vector.

File: tesseract_planning/src/trajopt/trajopt_freespace_planner.cpp [hunk 1]
```diff
@@ -115,7 +117,7 @@ bool TrajOptFreespacePlanner::solve(PlannerResponse& response, TrajOptFreespaceP
       // Add initial joint position constraint
       std::shared_ptr<JointPosTermInfo> jv = std::shared_ptr<JointPosTermInfo>(new JointPosTermInfo);
       jv->coeffs = std::vector<double>(pci.kin->numJoints(), 1.0);
-      jv->targets = end_position->joint_positions_;
+      jv->targets = std::vector<double>(end_position->joint_positions_.data(), end_position->joint_positions_.data() + end_position->joint_positions_.size());
       jv->first_step = pci.basic_info.n_steps - 1;
       jv->last_step = pci.basic_info.n_steps - 1;
       jv->name = "target_joint_position";
```
Comment (mpowelson) [inline tesseract_planning/src/trajopt/trajopt_freespace_planner.cpp]: I realize this is something I left out, but shouldn't there be a break here?

Comment (mpowelson) [reply tesseract_planning/src/trajopt/trajopt_freespace_planner.cpp]: I looks like I'm missing it in array planner as well

**Comment by Levi-Armstrong** [comment]
I will add it to this PR

**Comment by Levi-Armstrong** [comment]
@mpowelson Added the remaining changes

**Comment by Levi-Armstrong** [comment]
@mpowelson Clang passed

**Pull request merged**